fromObject does not check for nullables when mapping

I have a proto file that depends on `google/api/protobuf/descriptor` (I'm using definitions from `google-proto-files=3.0.2` package) and building it like so:
`protoc -I=./node_modules/google-proto-files/ -I=. --ts_opt=unary_rpc_promise=true --ts_opt=no_namespace --ts_out=./src/generated rpc.proto`

It does work and generates various ts files, but all `fromObject` methods that interact with arrays (those in the `descriptor` proto file) have an error that prevent me from "compiling" ts.

```log
src/generated/google/protobuf/descriptor.ts:28:19 - error TS2532: Object is possibly 'undefined'.

28             file: data.file.map(item => FileDescriptorProto.fromObject(item))
                     ~~~~~~~~~

src/generated/google/protobuf/descriptor.ts:233:27 - error TS2532: Object is possibly 'undefined'.

233             message_type: data.message_type.map(item => DescriptorProto.fromObject(item)),
                              ~~~~~~~~~~~~~~~~~
...
```

Adding conditional access `?.` and a default `?? []`  would already make it work

I'm just getting started with protoc so any recommendations are welcome :)

--- a/src/compiler/descriptor.ts
+++ b/src/compiler/descriptor.ts
@@ -25,7 +25,7 @@ export class FileDescriptorSet extends pb_1.Message {
         file?: ReturnType<typeof FileDescriptorProto.prototype.toObject>[];
     }): FileDescriptorSet {
         const message = new FileDescriptorSet({
-            file: data.file.map(item => FileDescriptorProto.fromObject(item))
+            file: (data.file?.map)(item => FileDescriptorProto.fromObject(item)) ?? []
         });
         return message;
     }
@@ -216,10 +216,10 @@ export class FileDescriptorProto extends pb_1.Message {
             dependency: data.dependency,
             public_dependency: data.public_dependency,
             weak_dependency: data.weak_dependency,
-            message_type: data.message_type.map(item => DescriptorProto.fromObject(item)),
-            enum_type: data.enum_type.map(item => EnumDescriptorProto.fromObject(item)),
-            service: data.service.map(item => ServiceDescriptorProto.fromObject(item)),
-            extension: data.extension.map(item => FieldDescriptorProto.fromObject(item))
+            message_type: (data.message_type?.map)(item => DescriptorProto.fromObject(item)) ?? [],
+            enum_type: (data.enum_type?.map)(item => EnumDescriptorProto.fromObject(item)) ?? [],
+            service: (data.service?.map)(item => ServiceDescriptorProto.fromObject(item)) ?? [],
+            extension: (data.extension?.map)(item => FieldDescriptorProto.fromObject(item)) ?? []
         });
         if (data.name != null) {
             message.name = data.name;
@@ -483,13 +483,13 @@ export class DescriptorProto extends pb_1.Message {
         reserved_name: string[];
     }): DescriptorProto {
         const message = new DescriptorProto({
-            field: data.field.map(item => FieldDescriptorProto.fromObject(item)),
-            extension: data.extension.map(item => FieldDescriptorProto.fromObject(item)),
-            nested_type: data.nested_type.map(item => DescriptorProto.fromObject(item)),
-            enum_type: data.enum_type.map(item => EnumDescriptorProto.fromObject(item)),
-            extension_range: data.extension_range.map(item => DescriptorProto.ExtensionRange.fromObject(item)),
-            oneof_decl: data.oneof_decl.map(item => OneofDescriptorProto.fromObject(item)),
-            reserved_range: data.reserved_range.map(item => DescriptorProto.ReservedRange.fromObject(item)),
+            field: (data.field?.map)(item => FieldDescriptorProto.fromObject(item)) ?? [],
+            extension: (data.extension?.map)(item => FieldDescriptorProto.fromObject(item)) ?? [],
+            nested_type: (data.nested_type?.map)(item => DescriptorProto.fromObject(item)) ?? [],
+            enum_type: (data.enum_type?.map)(item => EnumDescriptorProto.fromObject(item)) ?? [],
+            extension_range: (data.extension_range?.map)(item => DescriptorProto.ExtensionRange.fromObject(item)) ?? [],
+            oneof_decl: (data.oneof_decl?.map)(item => OneofDescriptorProto.fromObject(item)) ?? [],
+            reserved_range: (data.reserved_range?.map)(item => DescriptorProto.ReservedRange.fromObject(item)) ?? [],
             reserved_name: data.reserved_name
         });
         if (data.name != null) {
@@ -860,7 +860,7 @@ export class ExtensionRangeOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): ExtensionRangeOptions {
         const message = new ExtensionRangeOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         return message;
     }
@@ -1423,8 +1423,8 @@ export class EnumDescriptorProto extends pb_1.Message {
         reserved_name: string[];
     }): EnumDescriptorProto {
         const message = new EnumDescriptorProto({
-            value: data.value.map(item => EnumValueDescriptorProto.fromObject(item)),
-            reserved_range: data.reserved_range.map(item => EnumDescriptorProto.EnumReservedRange.fromObject(item)),
+            value: (data.value?.map)(item => EnumValueDescriptorProto.fromObject(item)) ?? [],
+            reserved_range: (data.reserved_range?.map)(item => EnumDescriptorProto.EnumReservedRange.fromObject(item)) ?? [],
             reserved_name: data.reserved_name
         });
         if (data.name != null) {
@@ -1778,7 +1778,7 @@ export class ServiceDescriptorProto extends pb_1.Message {
         options?: ReturnType<typeof ServiceOptions.prototype.toObject>;
     }): ServiceDescriptorProto {
         const message = new ServiceDescriptorProto({
-            method: data.method.map(item => MethodDescriptorProto.fromObject(item))
+            method: (data.method?.map)(item => MethodDescriptorProto.fromObject(item)) ?? []
         });
         if (data.name != null) {
             message.name = data.name;
@@ -2350,7 +2350,7 @@ export class FileOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): FileOptions {
         const message = new FileOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.java_package != null) {
             message.java_package = data.java_package;
@@ -2717,7 +2717,7 @@ export class MessageOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): MessageOptions {
         const message = new MessageOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.message_set_wire_format != null) {
             message.message_set_wire_format = data.message_set_wire_format;
@@ -2913,7 +2913,7 @@ export class FieldOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): FieldOptions {
         const message = new FieldOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.ctype != null) {
             message.ctype = data.ctype;
@@ -3061,7 +3061,7 @@ export class OneofOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): OneofOptions {
         const message = new OneofOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         return message;
     }
@@ -3153,7 +3153,7 @@ export class EnumOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): EnumOptions {
         const message = new EnumOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.allow_alias != null) {
             message.allow_alias = data.allow_alias;
@@ -3255,7 +3255,7 @@ export class EnumValueOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): EnumValueOptions {
         const message = new EnumValueOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.deprecated != null) {
             message.deprecated = data.deprecated;
@@ -3345,7 +3345,7 @@ export class ServiceOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): ServiceOptions {
         const message = new ServiceOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.deprecated != null) {
             message.deprecated = data.deprecated;
@@ -3449,7 +3449,7 @@ export class MethodOptions extends pb_1.Message {
         uninterpreted_option?: ReturnType<typeof UninterpretedOption.prototype.toObject>[];
     }): MethodOptions {
         const message = new MethodOptions({
-            uninterpreted_option: data.uninterpreted_option.map(item => UninterpretedOption.fromObject(item))
+            uninterpreted_option: (data.uninterpreted_option?.map)(item => UninterpretedOption.fromObject(item)) ?? []
         });
         if (data.deprecated != null) {
             message.deprecated = data.deprecated;
@@ -3628,7 +3628,7 @@ export class UninterpretedOption extends pb_1.Message {
         aggregate_value?: string;
     }): UninterpretedOption {
         const message = new UninterpretedOption({
-            name: data.name.map(item => UninterpretedOption.NamePart.fromObject(item))
+            name: (data.name?.map)(item => UninterpretedOption.NamePart.fromObject(item)) ?? []
         });
         if (data.identifier_value != null) {
             message.identifier_value = data.identifier_value;
@@ -3855,7 +3855,7 @@ export class SourceCodeInfo extends pb_1.Message {
         location?: ReturnType<typeof SourceCodeInfo.Location.prototype.toObject>[];
     }): SourceCodeInfo {
         const message = new SourceCodeInfo({
-            location: data.location.map(item => SourceCodeInfo.Location.fromObject(item))
+            location: (data.location?.map)(item => SourceCodeInfo.Location.fromObject(item)) ?? []
         });
         return message;
     }
@@ -4070,7 +4070,7 @@ export class GeneratedCodeInfo extends pb_1.Message {
         annotation?: ReturnType<typeof GeneratedCodeInfo.Annotation.prototype.toObject>[];
     }): GeneratedCodeInfo {
         const message = new GeneratedCodeInfo({
-            annotation: data.annotation.map(item => GeneratedCodeInfo.Annotation.fromObject(item))
+            annotation: (data.annotation?.map)(item => GeneratedCodeInfo.Annotation.fromObject(item)) ?? []
         });
         return message;
     }

--- a/src/compiler/plugin.ts
+++ b/src/compiler/plugin.ts
@@ -212,7 +212,7 @@ export class CodeGeneratorRequest extends pb_1.Message {
     }): CodeGeneratorRequest {
         const message = new CodeGeneratorRequest({
             file_to_generate: data.file_to_generate,
-            proto_file: data.proto_file.map(item => dependency_1.FileDescriptorProto.fromObject(item))
+            proto_file: (data.proto_file?.map)(item => dependency_1.FileDescriptorProto.fromObject(item)) ?? []
         });
         if (data.parameter != null) {
             message.parameter = data.parameter;
@@ -336,7 +336,7 @@ export class CodeGeneratorResponse extends pb_1.Message {
         file?: ReturnType<typeof CodeGeneratorResponse.File.prototype.toObject>[];
     }): CodeGeneratorResponse {
         const message = new CodeGeneratorResponse({
-            file: data.file.map(item => CodeGeneratorResponse.File.fromObject(item))
+            file: (data.file?.map)(item => CodeGeneratorResponse.File.fromObject(item)) ?? []
         });
         if (data.error != null) {
             message.error = data.error;

--- a/src/descriptor.ts
+++ b/src/descriptor.ts
@@ -188,10 +188,17 @@ function createFromObject(
             [ts.factory.createIdentifier("item")],
           ),
         );
-        assignmentExpr = ts.factory.createCallExpression(
-          ts.factory.createPropertyAccessExpression(assignmentExpr, "map"),
-          undefined,
-          [arrowFunc],
+        assignmentExpr = ts.factory.createBinaryExpression(
+          ts.factory.createCallExpression(
+            ts.factory.createPropertyAccessChain(assignmentExpr, ts.factory.createToken(ts.SyntaxKind.QuestionDotToken), "map"),
+            undefined,
+            [arrowFunc],
+          ),
+          ts.factory.createToken(ts.SyntaxKind.QuestionQuestionToken),
+          ts.factory.createArrayLiteralExpression(
+            [],
+            false
+          )
         );
       } else {
         assignmentExpr = ts.factory.createCallExpression(

--- a/test/conformance/packedproto2/packed_proto2.ts
+++ b/test/conformance/packedproto2/packed_proto2.ts
@@ -1231,7 +1231,7 @@ export class InteractiveAnnotation extends pb_1.Message {
         location?: ReturnType<typeof Location.prototype.toObject>;
     }): InteractiveAnnotation {
         const message = new InteractiveAnnotation({
-            polygonVertices: data.polygonVertices.map(item => Point.fromObject(item))
+            polygonVertices: (data.polygonVertices?.map)(item => Point.fromObject(item)) ?? []
         });
         if (data.location != null) {
             message.location = Location.fromObject(data.location);
@@ -2260,7 +2260,7 @@ export class ImageMessage extends pb_1.Message {
         midQualityFileEncSha256?: Uint8Array;
     }): ImageMessage {
         const message = new ImageMessage({
-            interactiveAnnotations: data.interactiveAnnotations.map(item => InteractiveAnnotation.fromObject(item)),
+            interactiveAnnotations: (data.interactiveAnnotations?.map)(item => InteractiveAnnotation.fromObject(item)) ?? [],
             scanLengths: data.scanLengths
         });
         if (data.url != null) {
@@ -4385,7 +4385,7 @@ export class VideoMessage extends pb_1.Message {
         gifAttribution?: VideoMessage.VIDEO_MESSAGE_ATTRIBUTION;
     }): VideoMessage {
         const message = new VideoMessage({
-            interactiveAnnotations: data.interactiveAnnotations.map(item => InteractiveAnnotation.fromObject(item))
+            interactiveAnnotations: (data.interactiveAnnotations?.map)(item => InteractiveAnnotation.fromObject(item)) ?? []
         });
         if (data.url != null) {
             message.url = data.url;
@@ -5297,7 +5297,7 @@ export class ContactsArrayMessage extends pb_1.Message {
         contextInfo?: ReturnType<typeof ContextInfo.prototype.toObject>;
     }): ContactsArrayMessage {
         const message = new ContactsArrayMessage({
-            contacts: data.contacts.map(item => ContactMessage.fromObject(item))
+            contacts: (data.contacts?.map)(item => ContactMessage.fromObject(item)) ?? []
         });
         if (data.displayName != null) {
             message.displayName = data.displayName;
@@ -6145,7 +6145,7 @@ export class HighlyStructuredMessage extends pb_1.Message {
     }): HighlyStructuredMessage {
         const message = new HighlyStructuredMessage({
             params: data.params,
-            localizableParams: data.localizableParams.map(item => HSMLocalizableParameter.fromObject(item))
+            localizableParams: (data.localizableParams?.map)(item => HSMLocalizableParameter.fromObject(item)) ?? []
         });
         if (data.namespace != null) {
             message.namespace = data.namespace;
@@ -7585,7 +7585,7 @@ export class FourRowTemplate extends pb_1.Message {
         locationMessage?: ReturnType<typeof LocationMessage.prototype.toObject>;
     }): FourRowTemplate {
         const message = new FourRowTemplate({
-            buttons: data.buttons.map(item => TemplateButton.fromObject(item))
+            buttons: (data.buttons?.map)(item => TemplateButton.fromObject(item)) ?? []
         });
         if (data.content != null) {
             message.content = HighlyStructuredMessage.fromObject(data.content);
@@ -7883,7 +7883,7 @@ export class HydratedFourRowTemplate extends pb_1.Message {
         locationMessage?: ReturnType<typeof LocationMessage.prototype.toObject>;
     }): HydratedFourRowTemplate {
         const message = new HydratedFourRowTemplate({
-            hydratedButtons: data.hydratedButtons.map(item => HydratedTemplateButton.fromObject(item))
+            hydratedButtons: (data.hydratedButtons?.map)(item => HydratedTemplateButton.fromObject(item)) ?? []
         });
         if (data.hydratedContentText != null) {
             message.hydratedContentText = data.hydratedContentText;
@@ -11095,7 +11095,7 @@ export class TabletNotificationsInfo extends pb_1.Message {
         notifyMessage?: ReturnType<typeof NotificationMessageInfo.prototype.toObject>[];
     }): TabletNotificationsInfo {
         const message = new TabletNotificationsInfo({
-            notifyMessage: data.notifyMessage.map(item => NotificationMessageInfo.fromObject(item))
+            notifyMessage: (data.notifyMessage?.map)(item => NotificationMessageInfo.fromObject(item)) ?? []
         });
         if (data.timestamp != null) {
             message.timestamp = data.timestamp;
@@ -11385,7 +11385,7 @@ export class WebNotificationsInfo extends pb_1.Message {
         notifyMessages?: ReturnType<typeof WebMessageInfo.prototype.toObject>[];
     }): WebNotificationsInfo {
         const message = new WebNotificationsInfo({
-            notifyMessages: data.notifyMessages.map(item => WebMessageInfo.fromObject(item))
+            notifyMessages: (data.notifyMessages?.map)(item => WebMessageInfo.fromObject(item)) ?? []
         });
         if (data.timestamp != null) {
             message.timestamp = data.timestamp;

--- a/test/default/default_proto2.ts
+++ b/test/default/default_proto2.ts
@@ -339,7 +339,7 @@ export class DefaultMessageV2WithoutDefault extends pb_1.Message {
             map_string_string: new Map(Object.entries(data.map_string_string)),
             map_string_message: new Map(Object.entries(data.map_string_message).map(([key, value]) => [key, dependency_1.DefaultCommonMessage.fromObject(value)])),
             array_int32: data.array_int32,
-            array_message: data.array_message.map(item => dependency_1.DefaultCommonMessage.fromObject(item)),
+            array_message: (data.array_message?.map)(item => dependency_1.DefaultCommonMessage.fromObject(item)) ?? [],
             bytes: data.bytes
         });
         if (data.one_of_int32 != null) {

--- a/test/default/default_proto3.ts
+++ b/test/default/default_proto3.ts
@@ -373,7 +373,7 @@ export class DefaultMessageV3 extends pb_1.Message {
             message.array_int32 = data.array_int32;
         }
         if (data.array_message != null) {
-            message.array_message = data.array_message.map(item => dependency_1.DefaultCommonMessage.fromObject(item));
+            message.array_message = (data.array_message?.map)(item => dependency_1.DefaultCommonMessage.fromObject(item)) ?? [];
         }
         if (data.one_of_int32 != null) {
             message.one_of_int32 = data.one_of_int32;

--- a/test/messagefields.ts
+++ b/test/messagefields.ts
@@ -45,7 +45,7 @@ export class MessageFields extends pb_1.Message {
             message.sub_message = SubMessage.fromObject(data.sub_message);
         }
         if (data.array_prop != null) {
-            message.array_prop = data.array_prop.map(item => SubMessage.fromObject(item));
+            message.array_prop = (data.array_prop?.map)(item => SubMessage.fromObject(item)) ?? [];
         }
         return message;
     }

--- a/test/no_namespace/nested.ts
+++ b/test/no_namespace/nested.ts
@@ -62,7 +62,7 @@ export class SchedulingContext extends pb_1.Message {
     }): SchedulingContext {
         const message = new SchedulingContext({});
         if (data.env != null) {
-            message.env = data.env.map(item => SchedulingContextEnv.fromObject(item));
+            message.env = (data.env?.map)(item => SchedulingContextEnv.fromObject(item)) ?? [];
         }
         if (data.timeout != null) {
             message.timeout = data.timeout;

--- a/test/rpcs.ts
+++ b/test/rpcs.ts
@@ -528,7 +528,7 @@ export namespace Query {
         }): Result {
             const message = new Result({});
             if (data.objects != null) {
-                message.objects = data.objects.map(item => _Object.fromObject(item));
+                message.objects = (data.objects?.map)(item => _Object.fromObject(item)) ?? [];
             }
             return message;
         }

--- a/test/web/client/weather_checked.ts
+++ b/test/web/client/weather_checked.ts
@@ -252,7 +252,7 @@ export namespace weather {
             }): Result {
                 const message = new Result({});
                 if (data.cities != null) {
-                    message.cities = data.cities.map(item => City.fromObject(item));
+                    message.cities = (data.cities?.map)(item => City.fromObject(item)) ?? [];
                 }
                 return message;
             }

--- a/test/web/client/weather_promise_checked.ts
+++ b/test/web/client/weather_promise_checked.ts
@@ -252,7 +252,7 @@ export namespace weather {
             }): Result {
                 const message = new Result({});
                 if (data.cities != null) {
-                    message.cities = data.cities.map(item => City.fromObject(item));
+                    message.cities = (data.cities?.map)(item => City.fromObject(item)) ?? [];
                 }
                 return message;
             }

